Comments jm (#7)
pat-s/2019-feature-selection: docs/00-manuscripts/ieee/ieee-manuscript.tex

The \textit{Pinus radiata} plots of this study, namely Laukiz1, Laukiz2, Luiando and Oiartzun, are located in the northern part of the Basque Country (\autoref{fig:study_area}).
Oiartzun has the most observations (n = 559) while Laukiz2 shows the largest area size (1.44 ha).
All plots besides Luiando are located within 100 km from the coast (\autoref{fig:study_area}).

> In total 1808 observations are available (Laukiz1 = 559, Laukiz2 = 451, Luiando = 301, Oiartzun = 497).

Reviewer: Good catch!

\subsection{Hyperspectral data}

> The airborne hyperspectral data was acquired during two flight campaigns which took at noon on September 28th and October 5th 2016.

Reviewer: ```suggestion
The airborne hyperspectral data was acquired during two flight campaigns which took place at noon on September 28th and October 5th 2016.
```

\subsection{Feature selection}

> High-dimensional, feature-rich datasets come with several challenges for both model fitting and evaluation.

Reviewer: Good catch! Damn this happens quite often :/ and spell checks do not catch it

To reduce the feature space of a dataset, conceptually differing approaches exist: wrapper methods, filters, penalization methods (lasso and ridge) or \ac{PCA} \cite{bommert2020, das2001, guyon2003, jolliffe2016}.
In contrast to wrapper methods, filters can be added to the hyperparameter optimization step and have a lower computational footprint.

>  Due to the focus on filter methods in this manuscript, only this sub-group of feature selection methods will be introduced in greater detail in the following subsections.

Reviewer: ```suggestion
Due to the focus on filter methods in this manuscript, only this sub-group of feature selection methods will be introduced in greater detail in the following subsections.
```

The goal is to optimize the number of ranked features to the point at which the model achieves the best performance.

% Ensemble filter methods
Besides the concept of choosing a specific filter method to rank variables, studies showed that combining several filters using statistical operations such as 'minimum' or 'mean' can enhance the predictive performance of the resulting models, especially when applied to multiple datasets \cite{abeel2010, drotar2017a}.
This approach is referred to as 'ensemble filtering' \cite{dietterich2000}.
Ensemble filters align with the recent rise of the 'ensemble' approach in machine learning which uses the idea of stacking to combine the predictions of multiple models, aiming to enhance predictive performance \cite{polikar2012, feurer2015, bolon-canedo2019}.

> % JM: I have no idea what the Borda ensemble filter is. Maybe add one more sentence on this specific filter

Reviewer: My idea was that the reference and the sentence below is enough intro for this filter given that it does not play a major role down the line. What do you think?

In this work the 'Borda' ensemble filter was applied \cite{drotar2017a}.
Its final feature order is determined by the sum of all single filters ranks.

% Ensuring a fair weighting in the ensemble

> % JM: reading further on, it might be that there are only three classes, please refine, Patrick!

Reviewer: You are right that is confusing. There are three classes and each has one exclusive option in this case. I should make that more clear.

> Filter methods can be grouped into four classes: Correlation-based, entropy-based, linear and non-linear methods.

Reviewer: ```suggestion
Filter methods can be grouped into groups which are formed out of three binary classes: multivariate or univariate feature use, correlation or entropy-based importance weighting and linear and non-linear filter methodology.
```

	\item Normalized Ratio Indices (\ac{NRI}s): data-driven feature engineering)
\end{itemize}

> The idea of splitting the features into different sets originated from the question whether feature-engineered indices from reflectance values have a positive effect on model performance, as for example demonstrated by \cite{pena2017} within a spectro-temporal setting.

Reviewer: ```suggestion
The idea of splitting the features into different sets originated from the question whether feature-engineered indices derived from reflectance values have a positive effect on model performance.
\cite{pena2017} is an exemplary study which used this approach in a spectro-temporal setting.
```

When removing features with a correlation of nearly 1, lasso is able to score performances around 31 p.p. and shows, somewhat surprisingly, the best performance on the VI task.
The same general conclusion applies to ridge if one discards the problematic results on four tasks.
Regarding the problematic predictions of the ridge learner, a careful inspection of the hyperparameter optimization procedure of the fitted models ensured again that everything was correctly implemented.
One observation turned out to be influential in causing the outlier prediction value even though careful multivariate exploratory data analysis showed no reason for flagging this observation a priori.
It was therefore left in the model, but it should be acknowledged that this learner seemed particularly sensitive to the input data.

A potential limiting factor in this study could be the upper limit of 600 iterations used for the XGBoost algorithm (hyperparameter \texttt{nrounds}), especially for feature sets including NRIs (\autoref{tab:hyperparameter_limits}).
This setting was a compromise between runtime and tuning space extension with the goal to work well for most feature sets.

> It may be recommendable to increase this upper limit to a value closer to the number of features in the dataset in order to be able to exploit the full range of this hyperparameter.

Reviewer: ```suggestion
It may be recommendable to increase this upper limit to a value closer to the number of features in the dataset in order to be able to exploit the full potential of this hyperparameter.
```

\subsubsection{Feature set differences}

> One objective of ths study was whether expert-based or data-driven feature engineering has a positive influence on model performance.

Reviewer: ```suggestion
One objective of this study was whether expert-based or data-driven feature engineering has a positive influence on model performance.
```
